src/result.rs

```diff
@@ -147,530 +158,32 @@ impl<T, E: std::error::Error, S: fmt::Display> Termination for Result<T, E, S> {
-    /// ```
-    #[inline]
-    #[track_caller]
-    pub fn expect_err(self, msg: &str) -> E {
-        match self {
-            Ok(t) => unwrap_failed(msg, &t),
-            Err(e) => e.error,
-        }
-    }
-
-    /// Returns the contained [`Err`] value, consuming the `self` value.
-    ///
-    /// # Panics
-    ///
-    /// Panics if the value is an [`Ok`], with a custom panic message provided
-    /// by the [`Ok`]'s value.
-    ///
-    /// # Examples
-    ///
-    /// ```should_panic
-    /// let x: Result<u32, &str> = Ok(2);
-    /// x.unwrap_err(); // panics with `2`
-    /// ```
-    ///
-    /// ```
-    /// let x: Result<u32, &str> = Result::new_err("emergency failure");
-    /// assert_eq!(x.unwrap_err(), "emergency failure");
-    /// ```
-    #[inline]
-    #[track_caller]
-    pub fn unwrap_err(self) -> E {
-        match self {
-            Ok(t) => unwrap_failed("called `Result::unwrap_err()` on an `Ok` value", &t),
-            Err(e) => e.error,
-        }
-    }
-}
-
-impl<T: Default, E> Result<T, E> {
-    /// Returns the contained [`Ok`] value or a default
-    ///
-    /// Consumes the `self` argument then, if [`Ok`], returns the contained
-    /// value, otherwise if [`Err`], returns the default value for that
-    /// type.
-    ///
-    /// # Examples
-    ///
-    /// Converts a string to an integer, turning poorly-formed strings
-    /// into 0 (the default value for integers). [`parse`] converts
-    /// a string to any other type that implements [`FromStr`], returning an
-    /// [`Err`] on error.
-    ///
-    /// ```
-    /// let good_year_from_input = "1909";
-    /// let bad_year_from_input = "190blarg";
-    /// let good_year = good_year_from_input.parse().unwrap_or_default();
-    /// let bad_year = bad_year_from_input.parse().unwrap_or_default();
-    ///
-    /// assert_eq!(1909, good_year);
-    /// assert_eq!(0, bad_year);
-    /// ```
-    ///
-    /// [`parse`]: str::parse
-    /// [`FromStr`]: crate::str::FromStr
-    #[inline]
-    pub fn unwrap_or_default(self) -> T {
-        match self {
-            Ok(x) => x,
-            Err(_) => Default::default(),
-        }
-    }
-}
-
-impl<T, E> Result<Option<T>, E> {
-    /// Transposes a `Result` of an `Option` into an `Option` of a `Result`.
-    ///
-    /// `Ok(None)` will be mapped to `None`.
-    /// `Ok(Some(_))` and `Err(_)` will be mapped to `Some(Ok(_))` and `Some(Err(_))`.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// #[derive(Debug, Eq, PartialEq)]
-    /// struct SomeErr;
-    ///
-    /// let x: Result<Option<i32>, SomeErr> = Ok(Some(5));
-    /// let y: Option<Result<i32, SomeErr>> = Some(Ok(5));
-    /// assert_eq!(x.transpose(), y);
-    /// ```
-    #[inline]
-    pub fn transpose(self) -> Option<Result<T, E>> {
-        match self {
-            Ok(Some(x)) => Some(Ok(x)),
-            Ok(None) => None,
-            Err(e) => Some(Err(e)),
+    pub fn unpack(self) -> (std::result::Result<T, E>, Option<S>) {
```

Comment:
Need help coming up with a good name for this operation.

`unwrap` is descriptive, but it is not a good choice because it means something totally different than `{Result, Option}::unwrap()`.

`unpack` is descriptive, but to me it's not different enough from `unwrap`.

`get` is not very descriptive, but it isn't easily confusable with other operations.

Reply:
unpack seems okay to me. 

Some alternatives I can think of

- unstack
- extract
- take_stack
- split_stack
- decompose

none of these are really better though, lol